tools/ktx/command_extract.cpp

```diff
@@ -23,38 +23,150 @@
 
 #include "png.imageio/lodepng.h"
 #include "astc-encoder/Source/tinyexr.h"
+#include "astc-encoder/Source/astcenc.h"
 
 // -------------------------------------------------------------------------------------------------
 
 namespace ktx {
 
+struct All_t {};
+static constexpr All_t all{};
+
+// Small utility class for selecting one specific uint or all (used for image selection via indices)
+struct Selector {
+    bool all = false;
+    uint32_t value = 0;
+
+    explicit Selector(All_t) : all(true) {}
+    explicit Selector(uint32_t value) : value(value) {}
+
+    Selector& operator=(All_t) & {
+        all = true;
+        value = 0;
+        return *this;
+    }
+
+    Selector& operator=(uint32_t v) & {
+        all = false;
+        value = v;
+        return *this;
+    }
+
+    friend bool operator==(All_t, const Selector& var) {
+        return var.all;
+    }
+    friend bool operator==(const Selector& var, All_t) {
+        return var.all;
+    }
+    friend bool operator!=(All_t, const Selector& var) {
+        return !var.all;
+    }
+    friend bool operator!=(const Selector& var, All_t) {
+        return !var.all;
+    }
+    friend bool operator>=(const Selector& var, uint32_t value) {
+        return !var.all && var.value >= value;
+    }
+    friend bool operator>=(uint32_t value, const Selector& var) {
+        return !var.all && value >= var.value;
+    }
+};
+
+static uint32_t value_or(const std::optional<Selector> opt, uint32_t fallback) {
+    return opt ? opt->value : fallback;
+}
+
+} // namespace ktx
+
+template<> struct fmt::formatter<ktx::Selector> : fmt::formatter<uint32_t> {
+    template <typename FormatContext>
+    auto format(const ktx::Selector& var, FormatContext& ctx) const -> decltype(ctx.out()) {
+        return var.all ?
+            formatter<std::string_view>{}.format("all", ctx) :
+            formatter<uint32_t>::format(var.value, ctx);
+    }
+};
+
+namespace ktx {
+
 // -------------------------------------------------------------------------------------------------
 
 /** @page ktxtools_extract ktx extract
 @~English
 
-Extracts a KTX2 file.
-
-@warning TODO Tools P5: This page is incomplete
+Extract selected images from a KTX2 file.
 
 @section ktxtools_extract_synopsis SYNOPSIS
-    ktx extract [options] @e input_file
+    ktx extract [option...] @e input_file @e output_path
 
 @section ktxtools_extract_description DESCRIPTION
+    @b ktx @b extract can extract one or multiple images from the KTX2 file specified as the
+    @e input_file argument and based on the format save them as RAW, EXR or PNG image files
```

Comment:
s/ based on the format save/, based on the format, save/

RAW is not an acronym so don't write in all caps. Use "Raw".